### DISCUSSION TITLE
Is it possible to do model adaptation?

### DISCUSSION THREAD

**Discussion by talentcoco** [openai/whisper#66] [Q&A] [answered]
Is it possible to add a corpus of specific words, names of places etc, technical jargon to improve accuracy?

**Comment by jongwook** [comment, marked as answer]
That is possible to some degree using prompt engineering; I added the `--initial_prompt` option in 2037b65, and you could do

```bash
whisper audio.mp3 --initial_prompt "So we were just talking about DALL·E"
```

to make the model more likely to output "DALL·E" rather than "Dolly".

**Comment by talentcoco** [comment]
Thanks so much!! Is it possible to "boost" specific words?  It will be very helpful as I don't know what the initial prompt will be. Ideally I would like to "boost" a few likely words e.g.  "DALL-E", "WALL-E", "R2-D2", and "HAL9000"

**Reply by jongwook** [reply]
Currently there's no interface for this other than giving the `initial_prompt` like the above; you could hack something with [logit biasing](https://aidungeon.medium.com/controlling-gpt-3-with-logit-bias-55866d593292), that effectively boosts the predicted probability of certain tokens. The `LogitFilter` class is designed to support this:

https://github.com/openai/whisper/blob/62fe7f1009a534986ac1d32a4aef8c244d029c28/whisper/decoding.py#L366-L370
